test/cipherlist_test.c

```diff
@@ -105,20 +105,23 @@ static const uint32_t default_ciphers_in_order[] = {
 # endif
 #endif  /* !OPENSSL_NO_TLS1_2 */
 
-#ifndef OPENSSL_NO_EC
+#if !defined(OPENSSL_NO_TLS1_2) || defined(OPENSSL_NO_TLS1_3)
```

Comment:
I'm probably just confusing myself, but could you explain why this is ```||``` and not ```&&``` -- if we have neither 1.2 nor 1.3, this cipher is still not useful, right?

Reply:
Pay careful attention to which terms have `!`. These ciphersuites are only applicable in <= TLS1.2. A problem occurs if TLSv1.2 is disabled, but TLSv1.3 is *enabled*. Even though TLSv1.0 and TLSv1.1 are enabled OpenSSL will not use them because of the TLSv1.2 "hole". So this guard says "if TLSv1.2 is enabled or TLSv1.3 is disabled then use these ciphersuites".

Reply:
Double negation is a [curseword]... I was thinking the same thing as @kaduk, took me a while to wrap my head around the condition...